internal/cli/cmd/dac/build/build.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	persesCMD "github.com/perses/perses/internal/cli/cmd"
+	"github.com/perses/perses/internal/cli/opt"
+	"github.com/perses/perses/internal/cli/output"
+	"github.com/spf13/cobra"
+)
+
+const (
+	outputFolderName = "built"
+	modeFile         = "file"
+	modeStdout       = "stdout"
+)
+
+// writeToFile writes data to a file
+func writeToFile(filePath string, data []byte) error {
+	file, err := os.Create(filePath)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	_, err = file.Write(data)
+	return err
+}
+
+type option struct {
+	persesCMD.Option
+	opt.FileOption
+	opt.DirectoryOption
+	opt.OutputOption
+	writer io.Writer
+	Mode   string
+}
+
+func (o *option) Complete(args []string) error {
+	if len(args) > 0 {
+		return fmt.Errorf("no args are supported by the command 'setup'")
+	}
+
+	// Complete the output only if it has been set by the user
+	if len(o.Output) > 0 {
+		if outputErr := o.OutputOption.Complete(); outputErr != nil {
+			return outputErr
+		}
+	} else {
+		// Put explicitely a value when not provided, as we use it for file generation in Execute()
+		o.Output = output.YAMLOutput
+	}
+
+	if o.Mode != modeFile && o.Mode != modeStdout {
+		return fmt.Errorf("invalid mode provided: must be either `file` or `stdout`")
+	}
+
+	return nil
+}
+
+func (o *option) Validate() error {
+	if o.File != "" {
+		return o.FileOption.Validate()
+	}
+	return o.DirectoryOption.Validate()
+}
+
+func (o *option) Execute() error {
+	if o.File != "" {
+		return o.processFile(o.File)
+	}
+	// Else it's a directory, thus walk it and process each file
+	err := filepath.Walk(o.Directory, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Process regular files only
+		if !info.IsDir() {
```

Comment:
```suggestion
if info.IsDir() {
  return nil
}
```

Like that anything that you would like to ignore are checked in the first place and then you have the actual business code and it's not "hidden" in a condition